airflow/hooks/hive_hooks.py

```diff
@@ -78,25 +79,66 @@ def run_cli(self, hql, schema=None):
                 if self.hive_cli_params:
                     hive_params_list = self.hive_cli_params.split()
                     hive_cmd.extend(hive_params_list)
-                logging.info(" ".join(hive_cmd))
+                if verbose:
+                    logging.info(" ".join(hive_cmd))
                 sp = subprocess.Popen(
                     hive_cmd,
                     stdout=subprocess.PIPE,
                     stderr=subprocess.STDOUT,
                     cwd=tmp_dir)
-                all_err = ''
                 self.sp = sp
                 stdout = ''
                 for line in iter(sp.stdout.readline, ''):
                     stdout += line
-                    logging.info(line.strip())
+                    if verbose:
+                        logging.info(line.strip())
                 sp.wait()
 
                 if sp.returncode:
-                    raise AirflowException(all_err)
+                    raise AirflowException(stdout)
 
                 return stdout
 
+
+    def test_hql(self, hql):
+        """
+        Test an hql statement using the hive cli and EXPLAIN
+
+        """
+        create, insert, other = [], [], []
+        for query in hql.split(';'):  # naive
+            query = query.lower().strip()
+            if query.startswith('create table'):
+                create.append(query)
+            elif query.startswith(('set', 'add jar', 'temporary')):
```

Comment:
Trailing spaces? maybe at least for `set `
`('set ', 'add jar ', 'temporary ')`